From cac55ce175bbf125500ab20055987d2211b4f3da Mon Sep 17 00:00:00 2001
From: Tejas Patel <tejas.patel1@einfochips.com>
Date: Tue, 19 Oct 2021 14:47:08 +0530
Subject: [PATCH 12/16] IMX8 THOR96: Added support of spi2 for zigbee

 1. Changed the pin muxing,
 2. Added GPIO configuration for supporting mgm111 zigbee module
 3. Removed conflicting pin from audio codec

Change-Id: Id64975a8bab986d79b9827bc479d883c101ca326
Signed-off-by: Tejas Patel <tejas.patel1@einfochips.com>
Signed-off-by: Tanvi Chauhan <tanvi.chauhan@einfochips.com>

diff --git a/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts b/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts
index 75dfd9d9b..26480402e 100644
--- a/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts
@@ -189,13 +189,6 @@
 	};
 };
 
-/* LS-SPI0 */
-&ecspi2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_ecspi2>;
-	status = "okay";
-};
-
 &csi1_bridge {
 	fsl,mipi-mode;
 	fsl,two-8bit-sensor-mode;
@@ -650,6 +643,21 @@
 	};
 };
 
+&ecspi2 {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_ecspi2>, <&pinctrl_zigbee_ctrl>;
+        fsl,spi-num-chipselects = <1>;
+        spi-max-frequency = <10000000>;
+        status = "okay";
+
+        spidev@0 {
+                compatible = "rohm,dh2228fv";
+                reg = <0>;
+                spi-max-frequency = <10000000>;
+                status = "okay";
+        };
+};
+
 /* SDIO */
 &usdhc1 {
 	#address-cells = <0x1>;
@@ -731,15 +739,6 @@
 		>;
 	};
 
-	pinctrl_ecspi2: ecspi2grp {
-		fsl,pins = <
-			MX8MQ_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK		0x16
-			MX8MQ_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI		0x16
-			MX8MQ_IOMUXC_ECSPI2_MISO_ECSPI2_MISO		0x16
-			MX8MQ_IOMUXC_ECSPI2_SS0_ECSPI2_SS0		0x16
-		>;
-	};
-
 	pinctrl_fec1: fec1grp {
 		fsl,pins = <
 			MX8MQ_IOMUXC_ENET_MDC_ENET1_MDC			0x4
@@ -933,6 +932,23 @@
 		>;
 	};
 
+        pinctrl_ecspi2: ecspi2grp {/* zigbee spi interface */
+                fsl,pins = <
+                        MX8MQ_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK  0x049
+                        MX8MQ_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI  0x049
+                        MX8MQ_IOMUXC_ECSPI2_MISO_ECSPI2_MISO  0x049
+                        MX8MQ_IOMUXC_GPIO1_IO08_GPIO1_IO8       0xd6 /*chip select*/
+                >;
+        };
+
+        pinctrl_zigbee_ctrl: zigbectrlgrp {
+                fsl,pins = <
+                        MX8MQ_IOMUXC_SPDIF_RX_GPIO5_IO4                 0x016   /*reset*/
+                        MX8MQ_IOMUXC_NAND_DATA05_GPIO3_IO11     0x016  /* Int */
+                        MX8MQ_IOMUXC_NAND_DATA04_GPIO3_IO10     0x016 /* wake*/
+                >;
+        };
+
 	pinctrl_ss_sel: usb3ssgrp{
 		fsl,pins = <
 			MX8MQ_IOMUXC_NAND_RE_B_GPIO3_IO15		0x16
@@ -942,7 +958,6 @@
 	pinctrl_spdif1: spdif1grp {
 		fsl,pins = <
 			MX8MQ_IOMUXC_SPDIF_TX_SPDIF1_OUT	0xd6
-			MX8MQ_IOMUXC_SPDIF_RX_SPDIF1_IN		0xd6 /*USED IN ZIGBEE*/
 		>;
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/imx8mq.dtsi b/arch/arm64/boot/dts/freescale/imx8mq.dtsi
index 9f15c03b3..8cd660a22 100755
--- a/arch/arm64/boot/dts/freescale/imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mq.dtsi
@@ -858,7 +858,7 @@
 			ecspi1: spi@30820000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
-				compatible = "fsl,imx8mq-ecspi", "fsl,imx51-ecspi";
+				compatible = "fsl,imx8mq-ecspi", "fsl,imx51-ecspi", "fsl,imx6ul-ecspi";
 				reg = <0x30820000 0x10000>;
 				interrupts = <GIC_SPI 31 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clk IMX8MQ_CLK_ECSPI1_ROOT>,
@@ -870,7 +870,7 @@
 			ecspi2: spi@30830000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
-				compatible = "fsl,imx8mq-ecspi", "fsl,imx51-ecspi";
+				compatible = "fsl,imx8mq-ecspi", "fsl,imx51-ecspi", "fsl,imx6ul-ecspi";
 				reg = <0x30830000 0x10000>;
 				interrupts = <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clk IMX8MQ_CLK_ECSPI2_ROOT>,
-- 
2.17.1

